include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -270,9 +378,9 @@ return_value(bool __res, _Size __size, ::std::true_type)
     return __res ? 0 : __size;
 }
 
-template <typename _Size>
-_Size
-return_value(_Size __res, _Size __size, ::std::false_type)
+template <typename _Size1, typename _Size2>
+auto
+return_value(_Size1 __res, _Size2 __size, ::std::false_type)
```

Comment:
```suggestion
__return_value(_Size1 __res, _Size2 __size, ::std::false_type)
```
I think `return_value` is too generic name. It's better to uglify, though I understand that this is not a part of your modifications

Reply:
The comment is not actual any more. I've removed this helper function.